src/client/datascience/notebook/kernelProvider.ts

```diff
@@ -220,18 +228,32 @@ export class VSCodeKernelPickerProvider implements INotebookKernelProvider {
             );
         }
     }
-    private async getJupyterSessionManager() {
+    private async getJupyterSessionManager(resource: Uri) {
```

Comment:
I thought you did this in another change? Maybe my diffing isn't working. Seems unrelated to telemetry.

Reply:
Oh never mind. Saw the new telemetry at the bottom.

Reply:
That was different (separate PR to address kernel selection).
Here when things fail, i need the URI so we can log the errors (hence passing it into this).